cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/TemplatesClassloaderUtil.java

```diff
@@ -0,0 +1,315 @@
+package com.devonfw.cobigen.impl.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utilities related to the retrieval of Templates utility classes
+ */
+public class TemplatesClassloaderUtil {
+
+    /** Logger instance. */
+    private static final Logger LOG = LoggerFactory.getLogger(TemplatesClassloaderUtil.class);
+
+    /**
+     * Checks the ClassLoader for any context.xml provided either in configurationFolder or in
+     * templates-plugin and returns its URL
+     * @param classLoader
+     *            ClassLoader to check resources from
+     * @return URL of the context configuration file path
+     * @throws IOException
+     *             if no configuration file was found
+     */
+    public URL getContextConfiguration(ClassLoader classLoader) throws IOException {
+        URL contextConfigurationLocation = null;
+        String[] possibleLocations = new String[] { "context.xml", "src/main/templates/context.xml" };
+
+        for (String possibleLocation : possibleLocations) {
+            URL configLocation = classLoader.getResource(possibleLocation);
+            if (configLocation != null) {
+                contextConfigurationLocation = configLocation;
+                LOG.debug("Found context.xml URL in the classpath @ {}", contextConfigurationLocation);
+                break;
+            }
+        }
+
+        if (contextConfigurationLocation == null) {
+            throw new IOException("No context.xml could be found in the classpath!");
```

Comment:
Better create a configuration exception which is already available in cobigen which indicates a problem with the configuration and is properly handled by eclipse or maven plugin already

Reply:
Adjusted to InvalidConfigurationException.